This PR appears to cause test failures for Python 3.6 and NumPy 1.17.5:

```
$ pytest tests/lax_numpy_test.py -k testResize --tb=short
========================================= test session starts =========================================
platform darwin -- Python 3.6.12, pytest-6.1.2, py-1.9.0, pluggy-0.13.1
rootdir: /Users/phawkins/p/jaxlg, configfile: pytest.ini
plugins: xdist-2.1.0, forked-1.3.0
collected 4145 items / 4135 deselected / 10 selected

tests/lax_numpy_test.py .......F.F                                                              [100%]

============================================== FAILURES ===============================================
_______________ LaxBackedNumpyTests.testResize_inshape=float32[3,1]_outshape=float32[] ________________
tests/lax_numpy_test.py:3181: in testResize
    self._CheckAgainstNumpy(np_fun, jnp_fun, args_maker)
jax/test_util.py:991: in _CheckAgainstNumpy
    numpy_ans = numpy_reference_op(*args)
tests/lax_numpy_test.py:3178: in <lambda>
    np_fun = lambda x: np.resize(x, out_shape)
<__array_function__ internals>:6: in resize
    ???
../../.pyenv/versions/3.6.12/envs/py36/lib/python3.6/site-packages/numpy/core/fromnumeric.py:1372: in resize
    a = a[:-extra]
E   TypeError: slice indices must be integers or None or have an __index__ method
___________________ LaxBackedNumpyTests.testResize_inshape=int8[4]_outshape=int8[] ____________________
tests/lax_numpy_test.py:3181: in testResize
    self._CheckAgainstNumpy(np_fun, jnp_fun, args_maker)
jax/test_util.py:991: in _CheckAgainstNumpy
    numpy_ans = numpy_reference_op(*args)
tests/lax_numpy_test.py:3178: in <lambda>
    np_fun = lambda x: np.resize(x, out_shape)
<__array_function__ internals>:6: in resize
    ???
../../.pyenv/versions/3.6.12/envs/py36/lib/python3.6/site-packages/numpy/core/fromnumeric.py:1372: in resize
    a = a[:-extra]
E   TypeError: slice indices must be integers or None or have an __index__ method
======================================= short test summary info =======================================
FAILED tests/lax_numpy_test.py::LaxBackedNumpyTests::testResize_inshape=float32[3,1]_outshape=float32[]
FAILED tests/lax_numpy_test.py::LaxBackedNumpyTests::testResize_inshape=int8[4]_outshape=int8[] - Ty...
============================ 2 failed, 8 passed, 4135 deselected in 3.40s =============================
```

I'm not sure why this failure doesn't show up in CI.

--- a/docs/jax.numpy.rst
+++ b/docs/jax.numpy.rst
@@ -310,6 +310,7 @@ Not every function in NumPy is implemented; contributions are welcome!
     remainder
     repeat
     reshape
+    resize
     result_type
     right_shift
     rint

--- a/jax/_src/numpy/lax_numpy.py
+++ b/jax/_src/numpy/lax_numpy.py
@@ -50,7 +50,7 @@ from jax import lax
 from jax._src.lax.lax import _device_put_raw
 from jax import ops
 from jax._src.ops import scatter
-from jax._src.util import (partial, unzip2, prod as _prod, subvals, safe_zip,
+from jax._src.util import (partial, unzip2, prod as _prod, subvals, safe_zip, ceil_of_ratio,
                            canonicalize_axis as _canonicalize_axis, maybe_named_axis)
 from jax.tree_util import tree_leaves, tree_flatten, tree_map
 
@@ -1415,6 +1415,23 @@ def unravel_index(indices, shape):
   idx = clipped_indices % cumulative_sizes[:-1] // cumulative_sizes[1:]
   return tuple(idx)
 
+@_wraps(np.resize)
+def resize(a, new_shape):
+  new_shape = _ensure_index_tuple(new_shape)
+
+  if _any(dim_length < 0 for dim_length in new_shape):
+    raise ValueError("all elements of `new_shape` must be non-negative")
+
+  a = ravel(a)
+
+  new_size = _prod(new_shape)
+  if a.size == 0 or new_size == 0:
+    return zeros_like(a, shape=new_shape)
+
+  repeats = ceil_of_ratio(new_size, a.size)
+  a = tile(a, repeats)[:new_size]
+
+  return reshape(a, new_shape)
 
 @_wraps(np.squeeze)
 def squeeze(a, axis: Optional[Union[int, Tuple[int, ...]]] = None):

--- a/jax/numpy/__init__.py
+++ b/jax/numpy/__init__.py
@@ -54,7 +54,7 @@ from jax._src.numpy.lax_numpy import (
     pi, piecewise, poly, polyadd, polyder, polyint, polymul, polysub, polyval, positive, power,
     prod, product, promote_types, ptp, quantile,
     r_, rad2deg, radians, ravel, ravel_multi_index, real, reciprocal, remainder, repeat, reshape,
-    result_type, right_shift, rint, roll, rollaxis, rot90, round, round_, row_stack,
+    resize, result_type, right_shift, rint, roll, rollaxis, rot90, round, round_, row_stack,
     save, savez, searchsorted, select, set_printoptions, setdiff1d, setxor1d, shape, sign, signbit,
     signedinteger, sin, sinc, single, sinh, size, sometrue, sort, sort_complex, split, sqrt,
     square, squeeze, stack, std, subtract, sum, swapaxes, take, take_along_axis,

--- a/tests/lax_numpy_test.py
+++ b/tests/lax_numpy_test.py
@@ -3167,6 +3167,22 @@ class LaxBackedNumpyTests(jtu.JaxTestCase):
     self._CompileAndCheck(jnp_fun, args_maker)
 
   @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name": "_inshape={}_outshape={}".format(
+          jtu.format_shape_dtype_string(arg_shape, dtype),
+          jtu.format_shape_dtype_string(out_shape, dtype)),
+       "arg_shape": arg_shape, "out_shape": out_shape, "dtype": dtype}
+      for dtype in default_dtypes
+      for arg_shape, out_shape in itertools.product(all_shapes, array_shapes)))
+  def testResize(self, arg_shape, out_shape, dtype):
+    rng = jtu.rand_default(self.rng())
+    np_fun = lambda x: np.resize(x, out_shape)
+    jnp_fun = lambda x: jnp.resize(x, out_shape)
+    args_maker = lambda: [rng(arg_shape, dtype)]
+    if len(out_shape) > 0 or numpy_version >= (1, 20, 0):
+      self._CheckAgainstNumpy(np_fun, jnp_fun, args_maker)
+    self._CompileAndCheck(jnp_fun, args_maker)
+
+  @parameterized.named_parameters(jtu.cases_from_list(
       {"testcase_name": "_inshape={}_expanddim={!r}".format(
           jtu.format_shape_dtype_string(arg_shape, dtype), dim),
        "arg_shape": arg_shape, "dtype": dtype, "dim": dim}

